### PR TITLE
feat: added icons & illustrations build

### PR DESCRIPTION
**Pull request by cscheffauer**
# Description

- Adjusted the existing icons builder to be more abstract & named it assets builder
- Added file handler to assets builder
- Added svgo transform to the assets builder
- Added the assets-builder + config to the illustrations package (cause those are also svgs, which need to be transformed)
- Small adjustments to general files, like README.md, etc.

To test it locally:
1. Run `yarn` on root
2. Run `yarn builder build` on root
3. Run `yarn icons build` on root and then check the `dist` folder in icons package
4. Run `yarn illustrations build` on root and then check the `dist` folder in illustrations package

TODOs: 
- Add integration tests

## Links

https://github.com/svg/svgo

### CODE REVIEW

File: packages/@momentum-design/icons/config/momentum.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                    "params": {
+                        "overrides": {
+                            "removeViewBox": false
+                        }
+                    }
+                },
+                {
+                    "name": "removeAttrs",
+                    "params": {
+                      "attrs": "*:fill:none"
```
Comment (ArthusLiang) [inline packages/@momentum-design/icons/config/momentum.json:22]: core icons will remove fill and stoken,
But illustrations, colored icons and brand icons will keep them.

File: packages/@momentum-design/illustrations/config/momentum.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    "params": {
+                        "overrides": {
+                            "removeViewBox": false
+                        }
+                    }
+                },
+                {
+                    "name": "removeAttrs",
+                    "params": {
+                      "attrs": "*:fill:none"
```
Comment (ArthusLiang) [inline packages/@momentum-design/illustrations/config/momentum.json:22]: no need to remove fill in illustrations.

Comment (cscheffauer) [reply packages/@momentum-design/illustrations/config/momentum.json:22]: done

File: packages/@momentum-design/builder/src/assets/builder/builder.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  fileHandler: FileHandler;
+  transformer: Transformer;
+}
+
+/**
+ * The Assets Builder class.
+ *
+ * Contains initialising and processing functions and makes use of
+ * several utilities, like `FileHandler` & `Transformer` utils.
+ */
```
Comment (InteractiveTimmy) [inline packages/@momentum-design/builder/src/assets/builder/builder.ts:30]: Needs `release` type tag, see [this `beta` tag documentation](https://api-extractor.com/pages/tsdoc/tag_beta/)

File: packages/@momentum-design/builder/src/assets/builder/file-handler.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+const PACKAGE = 'builder';
+const logger = Logger.child(generateMetadata(PACKAGE, `${CONSTANTS.TYPE}-file-handler`));
+
+/**
+ * The FileHandler class.
+ *
+ * Contains file handling utilities necessary for IO operations.
+ */
```
Comment (InteractiveTimmy) [inline packages/@momentum-design/builder/src/assets/builder/file-handler.ts:23]: Needs `release` type tag, see [this `beta` tag documentation](https://api-extractor.com/pages/tsdoc/tag_beta/)

File: packages/@momentum-design/icons/package.json [hunk 5]
```diff
@@ -9,8 +9,8 @@
     "analyze": "yarn analyze:prebuild && yarn analyze:postbuild",
     "analyze:postbuild": "echo \"script 'analyze:postbuild' has not been implemented\"",
     "analyze:prebuild": "echo \"script 'analyze:prebuild' has not been implemented\"",
-    "build": "yarn build:core",
-    "build:core": "md-builder -- --definition --icons --config ./config/webex.json",
+    "build": "rimraf dist && yarn build:core",
```
Comment (InteractiveTimmy) [inline packages/@momentum-design/icons/package.json:12]: build and cleaning functionality should be separate, this allows for automation to call `clean` recursively using `yarn workspace {target} clean`. If you could move `rimraf dist` to a script called `clean:dist`, called by `clean` within the scope of this package, it should allow for these two operations to be performed appropriately from the top level.

A good example of how these scripts *could* be implemented is in the `builder` package.json file.

File: packages/@momentum-design/illustrations/package.json [hunk 6]
```diff
@@ -9,7 +9,8 @@
     "analyze": "yarn analyze:prebuild && yarn analyze:postbuild",
     "analyze:postbuild": "echo \"script 'analyze:postbuild' has not been implemented\"",
     "analyze:prebuild": "echo \"script 'analyze:prebuild' has not been implemented\"",
-    "build": "echo \"script 'build' has not been implemented\"",
+    "build": "rimraf dist && yarn build:core",
```
Comment (InteractiveTimmy) [inline packages/@momentum-design/illustrations/package.json:12]: same deal here as with the icons package, see the previous comment.

**Review by InteractiveTimmy — APPROVED** [review]
LGTM! Thank you for making the changes!

**Pull request merged**